integration-tests/bats/rebase.bats

```diff
@@ -0,0 +1,396 @@
+#!/usr/bin/env bats
+load $BATS_TEST_DIRNAME/helper/common.bash
+
+setup() {
+    setup_common
+    dolt sql -q "CREATE table t1 (pk int primary key, c int);"
+    dolt add t1
+    dolt commit -m "main commit 1"
+    dolt branch b1
+    dolt sql -q "INSERT INTO t1 VALUES (1,1);"
+    dolt add t1
+    dolt commit -m "main commit 2"
+
+    dolt checkout b1
+    dolt sql -q "CREATE table t2 (pk int primary key);"
+    dolt add t2
+    dolt commit -m "b1 commit 1"
+
+    dolt checkout main
+}
+
+teardown() {
+    assert_feature_version
+    teardown_common
+}
+
+@test "rebase: no rebase in progress errors" {
+    run dolt rebase --abort
+    [ "$status" -eq 1 ]
+    [[ "$output" =~ "no rebase in progress" ]] || false
+
+    run dolt rebase --continue
+    [ "$status" -eq 1 ]
+    [[ "$output" =~ "no rebase in progress" ]] || false
+}
+
+@test "rebase: -i flag required" {
+    run dolt rebase b1
+    [ "$status" -eq 1 ]
+    [[ "$output" =~ "non-interactive rebases not currently supported" ]] || false
+}
+
+@test "rebase: bad args" {
+    run dolt rebase -i
+    [ "$status" -eq 1 ]
+    [[ "$output" =~ "not enough args" ]] || false
+
+    run dolt rebase -i main b1
+    [ "$status" -eq 1 ]
+    [[ "$output" =~ "rebase takes at most one positional argument" ]] || false
+
+    run dolt rebase --abrot
+    [ "$status" -eq 1 ]
+    [[ "$output" =~ "error: unknown option \`abrot'" ]] || false
+
+    run dolt rebase -i foo
+    [ "$status" -eq 1 ]
+    [[ "$output" =~ "branch not found: foo" ]] || false
+}
+
+@test "rebase: cannot rebase with dirty working set" {
+    dolt sql -q "INSERT INTO t1 VALUES (2,2);"
+    run dolt rebase -i b1
+    [ "$status" -eq 1 ]
+    [[ "$output" =~ "cannot start a rebase with uncommitted changes" ]] || false
+}
+
+@test "rebase: cannot rebase during active merge" {
+    dolt checkout b1
+    dolt sql -q "INSERT INTO t1 VALUES (1,2);"
+    dolt add t1
+    dolt commit -m "b1 commit 2"
+
+    run dolt merge main
+    [ "$status" -eq 1 ]
+    [[ "$output" =~ "Automatic merge failed" ]] || false
+
+    run dolt rebase -i main
+    [ "$status" -eq 1 ]
+    [[ "$output" =~ "unable to start rebase while a merge is in progress – abort the current merge before proceeding" ]] || false
+}
+
+@test "rebase: rebase working branch already exists" {
+    dolt checkout b1
+    dolt branch dolt_rebase_b1
+
+    run dolt rebase -i main
+    [ "$status" -eq 1 ]
+    [[ "$output" =~ "fatal: A branch named 'dolt_rebase_b1' already exists." ]] || false
+}
+
+@test "rebase: verify custom script" {
+    touch rebaseScript.sh
+    echo "#!/bin/bash" >> rebaseScript.sh
+    echo "mv rebasePlan.txt \$1" >> rebaseScript.sh
+    chmod +x rebaseScript.sh
+    export EDITOR=$PWD/rebaseScript.sh
+    export DOLT_TEST_FORCE_OPEN_EDITOR="1"
```

Comment:
You might consider putting some of this code into a helper function. It looks like it's used nine times in the file, so if you're able to reuse 5 of these 6 lines and not have to copy them a bunch of times in this file, I think it'd be a nice win and keep each test just a little bit simpler. Imaging how this file will change over time... we'll probably need to add more tests, so that helper function will probably help more in the future, too.